src/network/message.rs

```diff
@@ -38,6 +38,10 @@ use crate::util::merkleblock::MerkleBlock;
 /// This limit is not currently enforced by this implementation.
 pub const MAX_INV_SIZE: usize = 50_000;
 
+/// Maximum size, in bytes, of a an encoded message
+/// This by neccessity should be larger tham `MAX_VEC_SIZE`
+pub const MAX_MSG_SIZE: usize = 5_000_000;
+
```

Comment:
In ae28a501e0e41a78afb6b01dcf1fba8a689262dd:

I think `MAX_MSG_SIZE` should be capped to the (4 meg) block size, while `MAX_VEC_SIZE` should be slightly larger, since it represents the in-memory footprint of Bitcoin objects, which are sometimes larger than their serialized representation. I also don't think we should say "by necessity" anymore since we have flexible allocations.

Reply:
I though that the message contains the block, so `MAX_MSG_SIZE + some prefixexes >= MAX_VEC_SIZE`?

> `MAX_VEC_SIZE` should be slightly larger, since it represents the in-memory footprint of Bitcoin objects, which are sometimes larger than their serialized representation.

Right now `MAX_VEC_SIZE` is used a limit on the amount of data we decode from the reader, so it strictly represents the serialized data size.

Reply:
Ah ok, I misunderstood what `MAX_MSG_SIZE` was. I thought it was a cap on the amount of memory used by any data structure, rather than a cap on the serialized size of a network message.

Ignore me.